### PR TITLE
Replace old expired apple certificate + simple devdocs for updating

### PR DESCRIPTION
**Pull request by gbaraldi**

### CODE REVIEW

**Comment by giordano** [comment]
> + simple devdocs for updating

That sounds great but...where is it? 😬

File: devdocs/macos-developer-id.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## Expired Apple Developer ID
+
```
Comment (giordano) [inline devdocs/macos-developer-id.md:2]: Can you please explain that a symptom of the certificate needing to be updated is an error in the macOS upload jobs on `master` looking something like
```
Codesigning file dmg/Julia-1.12.app/Contents/MacOS/applet with identity 2053E9292809B66582CA9F042B470C0929340362
error: The specified item could not be found in the keychain.
```
?  That was very unclear to me when I saw it the other day.

File: devdocs/macos-developer-id.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+1. To replace an expired Apple certificate, you must download the julia-buildkite repo and decrypt by running `make decrypt`.
```
Comment (giordano) [inline devdocs/macos-developer-id.md:3]:
```suggestion
1. To replace an expired Apple certificate, clone the https://github.com/JuliaCI/julia-buildkite repository and decrypt it by running `make decrypt`.
```
Also, might explain how to get the `decrypt` program:
```console
% make decrypt 
cd .buildkite/cryptic_repo_root && decrypt --repo-root=$(pwd) --verbose
/bin/sh: line 1: decrypt: command not found
make: *** [Makefile:4: decrypt] Error 127
```

Furthermore, from what I understand below one needs to do this on macOS, it'd be better to make it clear from the beginning.

File: devdocs/macos-developer-id.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+2. Get the macos_codesigning.keychain file and add it to your local keychains with the Keychain Access app.
```
Comment (giordano) [inline devdocs/macos-developer-id.md:4]:
```suggestion
2. Get the `macos_codesigning.keychain` file and add it to your local keychains with the Keychain Access app.
```
Get it from where?

File: devdocs/macos-developer-id.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+3. From that app delete the old certificate and add the new one (it's a .cer file).
```
Comment (giordano) [inline devdocs/macos-developer-id.md:5]:
```suggestion
3. From that app delete the old certificate and add the new one (it's a `.cer` file).
```

File: devdocs/macos-developer-id.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+## Expired Apple Developer ID
```
Comment (giordano) [inline devdocs/macos-developer-id.md:1]:
```suggestion
## Update expired Apple Developer ID
```

**Comment by giordano** [comment]
This looks much better to me now, thanks!

**Comment by IanButterworth** [comment]
Looks good. Merge?

**Pull request merged**